lib/report.js

```diff
@@ -158,7 +160,117 @@ class ConciseReporter extends Reporter {
   }
 }
 
+class TapReporter extends Reporter {
+  constructor(options) {
+    super(options);
+    this.successful = 0;
+    this.failed = 0;
+    this.skipped = 0;
+    this.failedTests = [];
+    this.testNumeration = 1;
```

Comment:
You use `this.testNumeration - 1` more often than just `this.testNumeration`, so I suggest replacing them with `this.testNumeration` and `this.testNumeration + 1` respectively and starting it from zero.
```suggestion
    this.testNumeration = 0;
```